src/_pytest/monkeypatch.py

```diff
@@ -111,8 +111,16 @@ def __repr__(self) -> str:
 
 @final
 class MonkeyPatch:
-    """Object returned by the ``monkeypatch`` fixture keeping a record of
-    setattr/item/env/syspath changes."""
+    """Helper to conveniently monkeypatch attributes/items/environment
+    variables/syspath.
+
+    Returned by the :fixture:`monkeypatch` fixture.
+
+    :versionchanged:: 6.2
+        Can now also be used directly as `pytest.MonkeyPatch()`, for when
+        the fixture is not available. In this case, use :meth:`context` or
+        remember to call :meth:`undo` explicitly.
```

Comment:
While we are exposing it now, how about we make it a context manager so users can use it with `with`?

```python
with MonkeyPatch() as mp:
    mp.setattr(...)
```

It would require only adding two methods:

```python
    def __enter__(self):
        return self
    def __exit__(self, *args):
        self.undo()   
```

(But of course you already know that)

This of course could also be done in a separate PR, but thought I would mention it because it is simple to do. 👍

Reply:
Im strongly opposed as the monkeypatch object is the under as well and accidentally misusing it seems to be too easy

It would be neat to turn the context method to a class method instead

That way the context manager could use `MonkeyPatch.context()`

Reply:
Hmm I only just now noticed that `MonkeyPatch.context` returns a *new* instance so can indeed become a classmethod. I thought it operated on the instance.

Agreed that making `MonkeyPatch` itself a context manager would probably be misused like this:

```py
def test_it(monkeypatch):
    with monkeypatch:
          monkeypatch.setattr('foo', 'bar')
```

This will undo also monkeypatches set by other fixtures etc. I don't think there's a way to make it work without a breaking change or internal reenterancy bookkeeping which is undesirable.

So I'll make the classmethod change. I think it also clarifies the behavior regardless of the export question.

Reply:
> Agreed that making MonkeyPatch itself a context manager would probably be misused like this

Ahh good point, didn't realize that.

> So I'll make the classmethod change. I think it also clarifies the behavior regardless of the export question.

Sounds good. 👍